src/postings/recorder.rs

```diff
@@ -278,7 +279,8 @@ impl Recorder for TfAndPositionRecorder {
             }
             if let Some(doc_id_map) = doc_id_map {
                 // this simple variant to remap may consume to much memory
-                doc_id_and_positions.push((doc_id_map.get_new_doc_id(doc), buffer_positions.to_vec()));
```

Comment:
Can you remind me why lending is not an option for doc_id_and_positions?

Reply:
lending of buffer_positions? this is just a temp vec which gets cleared on every loop